CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1903,9 +1903,18 @@ public static function submit($params) {
     $params['invoiceID'] = md5(uniqid(rand(), TRUE));
 
     // We want to move away from passing in amount as it is calculated by the actually-submitted params.
-    $params['amount'] = $params['amount'] ?? $form->getMainContributionAmount($params);
```

Comment:
The test was passing in the wrong amount because it was required - but it wasn't validated so we didn't notice is was wrong. By calculating amount based on price fields we get less brittle tests.

Note this function is accessed by ContributionPage.submit but not the UI form